### PR TITLE
Pagination component

### LINKED ISSUE #377: Pagination component
Related to #358 

Suspect something like this would be good:

```js
class PaginatedTable extends React.Component {
  render() {
    const Paginate = this.props.paginationComponent;
    const Table = this.props.tableComponent;

    return (
      <div>
        <Table
          columns={...}
          rows={...}
        />
        <Paginate 
          page={this.state.page}
          onChange={this.onPageChange}
        />
      </div>
    )
  }
}

/// then: 

<PaginatedTable
  columns={...}
  rows={...}
  tableComponent={Table}
  paginationComponent={Paginate}
/>
```

### PR DESCRIPTION
**Pull request by eliascodes**
Fixes #377 

#### Changes
* Adds `PaginatedTableWrapper` to both `Visits` page and `VisitorDetails` 
* Minor stuff, see commits

##### Adding New Features
* [x] Have I manually tested my new feature?
* [ ] ~~Is my new feature sufficiently well tested?~~
* [ ] ~~Have I updated the project documentation and README to reflect my changes?~~
* [x] Does the new feature conform to the desired project architecure?

### CODE REVIEW

**Comment by astroash** [comment]
Also a UX thing...

On the visitors details page, scrolling is internal to the component, rather than the page like the rest of the site. Not sure if this is a fixed height/overflow thing. I had a glance at the css and couldn't find what was causing this, and if it's easy to fix.

dp-07

File: client/src/cb-admin/components/PaginatedTableWrapper.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import Paginate from 'react-paginate';
+import styled from 'styled-components';
+
+const FullWidthWrapper = styled.div`
+  width: 100%;
+`;
+
+
+export default class PaginatedTableWrapper extends React.Component {
+  static propTypes = {
```
[the remark this answers is not in the document]
Comment (eliascodes) [reply client/src/cb-admin/components/PaginatedTableWrapper.js]: yes

File: client/src/cb-admin/components/PaginatedTableWrapper.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      PropTypes.element,
+      PropTypes.arrayOf(PropTypes.element),
+    ]).isRequired,
+  }
+
+  static defaultProps = {
+    rowsPerPage: 10,
+  }
+
+  static getDerivedStateFromProps(nextProps) {
```
[the remark this answers is not in the document]
Comment (eliascodes) [reply client/src/cb-admin/components/PaginatedTableWrapper.js]: Check the react 16 docs

File: client/src/cb-admin/components/PaginatedTableWrapper.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  onPageChange = ({ selected }) =>
+    this.setState(
+      { offset: Math.ceil(selected * this.props.rowsPerPage) },
+      this.loadRows,
+    )
+
+  loadRows = () => this.props.loadRows(this.state.offset)
+
+  render() {
+    const Table = React.Children.only(this.props.children);
```
[the remark this answers is not in the document]
Comment (eliascodes) [reply client/src/cb-admin/components/PaginatedTableWrapper.js]: Yes, prop types should be changed

**Comment by eliascodes** [comment]
@astroash you can raise the scrolling thing as a separate issue if you like, IMO it's not part of this PR.

File: client/src/cb-admin/components/PaginatedTableWrapper.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (astroash) [reply client/src/cb-admin/components/PaginatedTableWrapper.js]: I read [this](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops). Maybe my question should have been...

in this use case what's the benefit of using this method over componentDidMount? I can potentially imagine a scenario that the number of visits has grown whilst someone is clicking through pagination, then the `loadRows` will `setState` on the parent component passing down the new `full_count` _(👈 this should probs be camelCase)_, which would then be caught by this `getDerivedStateFromProps`...

However I'm unsure if I'm thinking about the flow correctly or if this was the intended purpose.

[the remark this answers is not in the document]
Comment (eliascodes) [reply client/src/cb-admin/components/PaginatedTableWrapper.js]: The scenario you describe is possible, your understanding of the flow seems correct.

**Pull request merged**